src/caret_analyze/architecture/architecture.py

```diff
@@ -101,16 +100,36 @@ def is_target_comm(comm: CommunicationStructValue):
         return Util.find_one(is_target_comm, self.communications)
 
     def get_path(self, path_name: str) -> PathStructValue:
-        return self._path_manager.get_named_path(path_name)
+        if path_name not in self.path_names:
+            raise InvalidArgumentError(f'Failed to get named path. {path_name} not exist.')
+
+        named_path: PathStruct = Util.find_one(lambda x: x._path_name == path_name, self._paths)
+        return named_path.to_value()
 
-    def add_path(self, path_name: str, path_info: PathStructValue) -> None:
-        self._path_manager.add_named_path(path_name, path_info)
+    def add_path(self, path_name: str, path_info: PathStruct) -> None:
+        if path_name in self.path_names:
+            raise InvalidArgumentError('Failed to add named path. Duplicate path name.')
+        named_path_info = PathStruct(path_name, path_info.child)
+        self._paths.append(named_path_info)
 
     def remove_path(self, path_name: str) -> None:
-        self._path_manager.remove_named_path(path_name)
+        if path_name not in self.path_names:
+            raise InvalidArgumentError(f'Failed to remove named path. {path_name} not exist.')
+
+        idx = None
+        for i, p in enumerate(self._paths):
+            if p.path_name == path_name:
+                idx = i
 
-    def update_path(self, path_name: str, path: PathStructValue) -> None:
-        self._path_manager.update_named_path(path_name, path)
+        if idx is not None:
+            self._paths.pop(idx)
+
+    def update_path(self, path_name: str, path: PathStruct) -> None:
```

Comment:
The API has changed.
Please try to absorb the changes within the Architecture class.

```suggestion
    def update_path(self, path_name: str, path: PathStructValue) -> None:
```

Reply:
Fixed it in [e671a20](https://github.com/tier4/CARET_analyze/pull/156/commits/e671a20faeb125d187fd543e00529ba9f5194119)